precompiles/staking/staking_test.go

```diff
@@ -0,0 +1,940 @@
+package staking
```

Comment:
Something we started doing more in unit test: using arrange-act-assert in comments: https://automationpanda.com/2020/07/07/arrange-act-assert-a-pattern-for-writing-good-tests/

Might make sense to use it here for better test clarity